torch/_torch_docs.py

```diff
@@ -993,60 +993,71 @@ def merge_dicts(*dicts):
-(see :func:`torch.set_default_tensor_type`) if `None` is given.
+    :func:`torch.tensor` creates a tensor that always copies the data from the input object.
 
-For example: NumPy arrays also implement the buffer protocol. However, since NumPy
-arrays have higher priority than objects implementing the buffer protocol, this function
-will handle them as NumPy arrays. In other words, it will infer its dtype as if using
-``torch.from_numpy`` (instead of ``torch.frombuffer``).
+    :func:`torch.from_numpy` creates a tensor that always shares memory from NumPy arrays.
 
-.. seealso::
-    :func:`torch.as_tensor` tries to avoid copies for tensors and NumPy arrays.
-    :func:`torch.tensor` always copies the data from the input object.
-    :func:`torch.from_numpy` creates a tensor that shares its memory with a NumPy array.
-    :func:`torch.frombuffer` creates a tensor that shares its memory with an object
-    that implements the buffer protocol.
-    :func:`torch.utils.dlpack.from_dlpack` creates a tensor that shares its memory
-    with the object represented in the dlpack.
+    :func:`torch.frombuffer` creates a tensor that always shares memory from objects that
+           implement the buffer protocol.
+
+    :func:`torch.utils.dlpack.from_dlpack` creates a tensor that always shares memory from
+           DLPack capsules.
 
 Args:
-    obj (object): a Python object that satisfies, at least, one of the five options
-           mentioned above.
+    obj (object): a tensor, NumPy array, DLPack Capsule, object that implements Python's
+           buffer protocol, scalar, or sequence of scalars.
 
 Keyword args:
-    dtype (:class:`torch.dtype`, optional): the desired data type of returned tensor.
-           Default: if ``None``, it will be inferred from :attr:`obj`.
-    copy (bool, optional): flags whether the object memory should be copied or not.
-           If ``None``, then the result tensor shares memory with the Python object
-           whenever possible. If ``True``, then the object memory is copied. If ``False``,
-           then the object memory is shared. If the object memory cannot be shared
-           and this flag is ``False``, then an error is thrown.
-    device (:class:`torch.device`, optional): the device of the constructed tensor.
-           If `None`, then the device of :attr:`obj` is used. Else, it either copies
-           the data, if :attr:`obj` lives in a different device, or it shares the
-           memory, if :attr:`obj` lives in the same device.
-    requires_grad (bool, optional): If autograd should record operations on the
-           returned tensor. However, if this flag is ``False`` and the input object
-           is a non-leaf :class:`Tensor`, this function will call :func:`torch.Tensor.detach`.
+    dtype (:class:`torch.dtype`, optional): the datatype of the returned tensor.
+           Default: ``None``, which causes the datatype of the returned tensor to be
+           inferred from :attr:`obj`.
```

Comment:
technically this affects the copy aspect of the behavior as mentioned above. might be worth re-iterating it here

Reply:
Hmm. Not sure how I can do that in few words, since it affects the result depending on the input (already explained in the function summary). For example, for objects that implement Python's buffer protocol, `dtype` indicates how the sequence of bytes are going to be interpreted (no copy will take place). What do you think?

Reply:
you can optionally add: 

As mentioned above, if :attr:`dtype` is not the same as :attr:`obj.dtype`, the returned tensor will not share its memory with :attr:`obj`

Reply:
That sounds good. But, do you think it is okay adding it, even though it's not necessarily true for all cases?
I mean, it's not necessarily true if:
1. `obj` is something that implements Python's Buffer Protocol (memory will be shared no matter the value of `dtype`)
2. `obj` is a Python scalar or list of scalars (memory won't be shared no matter the value of `dtype`)
Let me know what you think!

Reply:
oh I totally missed that! no let's skip then